tests/testthat/_snaps/match.md

```diff
@@ -153,33 +181,245 @@
       Each element of `foo` can match at most 1 observation from `bar`.
       x The element at location 1 has multiple matches.
 
-# `multiple` is validated
+# `relationship` handles one-to-one case
 
     Code
-      (expect_error(vec_locate_matches(1, 2, multiple = 1.5)))
+      (expect_error(vec_locate_matches(c(2, 1), c(1, 1), relationship = "one_to_one"))
+      )
+    Output
+      <error/vctrs_error_matches_relationship_one_to_one>
+      Error in `vec_locate_matches()`:
+      ! Each element can match at most 1 observation.
+      x The element at location 2 has multiple matches.
+    Code
+      (expect_error(vec_locate_matches(c(1, 1), c(1, 2), relationship = "one_to_one"))
+      )
+    Output
+      <error/vctrs_error_matches_relationship_one_to_one>
+      Error in `vec_locate_matches()`:
+      ! Each element can match at most 1 observation.
+      x The element at location 1 has multiple matches.
```

Comment:
Here is one place I think it is confusing that we have `needles_arg = ""` and `haystack_arg = ""` as defaults. One of these errors is related to `needles` and one is related to `haystack` but the errors are the same.

I think we should default them to `"needles"` and `"haystack"` and then completely disallow the empty string of `""` as a possible arg name in `vec_locate_matches()`. This would also greatly simplify my error generation code on the R side

Reply:
I think taking `""` args is probably a remnant of the early days before we had developed our approach to error reporting. We probably didn't have `error_call` arguments at the time, so we were hiding the call and it made sense to hide the arguments as well.

Now that we show the call, we should probably start naming the arguments by default as you suggest.